Tests/Packet++Test/Tests/WireGuardTests.cpp

```diff
@@ -0,0 +1,336 @@
+
+	// create WireGuard Handshake Initiation message
+	memcpy(origBuffer, buffer1, bufferLength1);
+
+	uint8_t expectedPublicKeyInit[32] = { 0x5f, 0xce, 0xc7, 0xc8, 0xe5, 0xc8, 0xe2, 0xe3, 0xf7, 0x98, 0x9e,
+		                                  0xef, 0x60, 0xc2, 0x28, 0xd8, 0x23, 0x29, 0xd6, 0x02, 0xb6, 0xb1,
+		                                  0xe2, 0xbb, 0x9d, 0x06, 0x8f, 0x89, 0xcf, 0x9d, 0x4d, 0x45 };
+
+	uint8_t expectedStaticKeyInit[48] = { 0x32, 0x78, 0x0f, 0x6d, 0x27, 0x26, 0x4f, 0x7b, 0x98, 0x70, 0x1f, 0xdc,
+		                                  0x27, 0xa4, 0xec, 0x00, 0xae, 0xb6, 0xbe, 0xcd, 0xbe, 0xf2, 0x33, 0x2f,
+		                                  0x1b, 0x40, 0x84, 0xca, 0xdb, 0x93, 0x82, 0x39, 0x35, 0xc0, 0x12, 0xae,
+		                                  0x25, 0x5e, 0x7b, 0x25, 0xef, 0xf1, 0x39, 0x40, 0xc3, 0x21, 0xfa, 0x6b };
+	uint8_t expectedTimestampInit[28] = { 0xd6, 0x6a, 0x2a, 0x87, 0xb0, 0x61, 0xdb, 0x14, 0x30, 0x17,
+		                                  0x3e, 0x93, 0x7f, 0x56, 0x93, 0x49, 0xde, 0x28, 0x56, 0xdc,
+		                                  0x5f, 0x26, 0x16, 0x76, 0x3e, 0xee, 0xaf, 0xc0 };
+
+	uint8_t expectedMac1Init[16] = { 0x53, 0x3b, 0x01, 0xdd, 0x96, 0x5e, 0x7e, 0xc7,
+		                             0x69, 0x76, 0xe2, 0x8f, 0x68, 0x3d, 0x67, 0x12 };
+
+	uint8_t expectedMac2Init[16] = { 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+		                             0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00 };
+
+	pcpp::WireGuardHandshakeInitiationLayer newHandshakeInitMessage(818952152, expectedPublicKeyInit,
+	                                                                expectedStaticKeyInit, expectedTimestampInit,
+	                                                                expectedMac1Init, expectedMac2Init);
+	pcpp::Packet wgHandshakeInitPacket(&rawPacket1);
+	pcpp::WireGuardHandshakeInitiationLayer* origHandshakeInitMessage =
+	    dynamic_cast<pcpp::WireGuardHandshakeInitiationLayer*>(wgHandshakeInitPacket.detachLayer(pcpp::WireGuard));
+	PTF_ASSERT_NOT_NULL(origHandshakeInitMessage);
+	PTF_ASSERT_EQUAL(newHandshakeInitMessage.getDataLen(), origHandshakeInitMessage->getDataLen());
+
+	PTF_ASSERT_EQUAL(newHandshakeInitMessage.getSenderIndex(), 818952152);
+	PTF_ASSERT_TRUE(memcmp(newHandshakeInitMessage.getInitiatorEphemeral().data(), expectedPublicKeyInit, 32) == 0);
+	PTF_ASSERT_TRUE(memcmp(newHandshakeInitMessage.getEncryptedInitiatorStatic().data(), expectedStaticKeyInit, 48) ==
+	                0);
+	PTF_ASSERT_TRUE(memcmp(newHandshakeInitMessage.getEncryptedTimestamp().data(), expectedTimestampInit, 28) == 0);
+	PTF_ASSERT_TRUE(memcmp(newHandshakeInitMessage.getMac1().data(), expectedMac1Init, 16) == 0);
+	PTF_ASSERT_TRUE(memcmp(newHandshakeInitMessage.getMac2().data(), expectedMac2Init, 16) == 0);
+
+	PTF_ASSERT_TRUE(wgHandshakeInitPacket.addLayer(&newHandshakeInitMessage));
+
+	PTF_ASSERT_EQUAL(wgHandshakeInitPacket.getRawPacket()->getRawDataLen(), bufferLength1);
```

Comment:
We can also test the full buffer:
```cpp
PTF_ASSERT_BUF_COMPARE(newHandshakeInitMessage.getData(), origHandshakeInitMessage->getData(), origHandshakeInitMessage->getDataLen());
```

This will ensure all the data in the new packet is correct.

Same goes for the other packets we create later in the test